fe/fe-core/src/main/java/org/apache/doris/nereids/trees/copier/ExpressionDeepCopier.java

```diff
@@ -64,13 +66,27 @@ public Expression visitAlias(Alias alias, DeepCopierContext context) {
     @Override
     public Expression visitSlotReference(SlotReference slotReference, DeepCopierContext context) {
         Map<ExprId, ExprId> exprIdReplaceMap = context.exprIdReplaceMap;
+        ExprId newExprId;
         if (exprIdReplaceMap.containsKey(slotReference.getExprId())) {
-            ExprId newExprId = exprIdReplaceMap.get(slotReference.getExprId());
-            return slotReference.withExprId(newExprId);
+            newExprId = exprIdReplaceMap.get(slotReference.getExprId());
         } else {
-            SlotReference newOne = new SlotReference(slotReference.getName(), slotReference.getDataType(),
-                    slotReference.nullable(), slotReference.getQualifier());
-            exprIdReplaceMap.put(slotReference.getExprId(), newOne.getExprId());
+            newExprId = StatementScopeIdGenerator.newExprId();
+            exprIdReplaceMap.put(slotReference.getExprId(), newExprId);
+        }
+        return slotReference.withExprId(newExprId);
+    }
+
+    @Override
+    public Expression visitArrayItemReference(ArrayItemReference arrayItemSlot, DeepCopierContext context) {
+        Map<ExprId, ExprId> exprIdReplaceMap = context.exprIdReplaceMap;
+        if (exprIdReplaceMap.containsKey(arrayItemSlot.getExprId())) {
+            ExprId newExprId = exprIdReplaceMap.get(arrayItemSlot.getExprId());
+            return new ArrayItemReference(newExprId, arrayItemSlot.getName(),
+                    arrayItemSlot.getArrayExpression());
```

Comment:
do we need rewrite the expression returned by arrayItemSlot.getArrayExpression()?